tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_CylindricalBlastWave.cpp

```diff
@@ -0,0 +1,369 @@
+  const double adiabatic_index = 1.3333333333333333333;
+
+  CylindricalBlastWaveProxy cylindrical_blast_wave(
+      inner_radius, outer_radius, inner_density, outer_density, inner_pressure,
+      outer_pressure, magnetic_field, adiabatic_index);
+
+  const double epsilon = 1.e-10;
+  Approx approx = Approx::custom().epsilon(epsilon * 100.0);
+  auto x =
+      make_with_value<tnsr::I<double, 3, Frame::Inertial>>(inner_radius, 0.0);
+
+  get<0>(x) = inner_radius;
+  CHECK(inner_density == get(get<hydro::Tags::RestMassDensity<double>>(
+                             cylindrical_blast_wave.grmhd_variables(x))));
+  get<0>(x) = inner_radius - epsilon;
+  CHECK(inner_density == get(get<hydro::Tags::RestMassDensity<double>>(
+                             cylindrical_blast_wave.grmhd_variables(x))));
+  get<0>(x) = inner_radius + epsilon;
+  CHECK_ITERABLE_CUSTOM_APPROX(inner_density,
+                               get(get<hydro::Tags::RestMassDensity<double>>(
+                                   cylindrical_blast_wave.grmhd_variables(x))),
+                               approx);
+
+  get<0>(x) = outer_radius;
+  CHECK(outer_density == get(get<hydro::Tags::RestMassDensity<double>>(
+                             cylindrical_blast_wave.grmhd_variables(x))));
+  get<0>(x) = outer_radius + epsilon;
+  CHECK(outer_density == get(get<hydro::Tags::RestMassDensity<double>>(
+                             cylindrical_blast_wave.grmhd_variables(x))));
+  get<0>(x) = outer_radius - epsilon;
+  CHECK_ITERABLE_CUSTOM_APPROX(outer_density,
+                               get(get<hydro::Tags::RestMassDensity<double>>(
+                                   cylindrical_blast_wave.grmhd_variables(x))),
+                               approx);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.PointwiseFunctions.AnalyticData.GrMhd.CylindricalBlastWave",
+    "[Unit][PointwiseFunctions]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{"PointwiseFunctions"};
+
+  test_create_from_options();
+  test_serialize();
+  test_move();
+
+  test_variables(std::numeric_limits<double>::signaling_NaN());
+  test_variables(DataVector(5));
+
+  test_density_on_and_near_boundaries();
+}
+
+struct BlastWave {
+  using type = grmhd::AnalyticData::CylindricalBlastWave;
+  static constexpr OptionString help = {
+      "Cylindrical blast wave analytic initial data."};
+};
+
+// [[OutputRegex, CylindricalBlastWave expects InnerRadius < OuterRadius]]
+[[noreturn]] SPECTRE_TEST_CASE(
+    "Unit.PointwiseFunctions.AnalyticData.GrMhd.CylBlastWaveInRadLess",
+    "[Unit][PointwiseFunctions]") {
+  ERROR_TEST();
+  grmhd::AnalyticData::CylindricalBlastWave cylindrical_blast_wave(
+      1.2, 1.0, 1.0e-2, 1.0e-4, 1.0, 5.0e-4,
+      std::array<double, 3>{{0.1, 0.0, 0.0}}, 1.3333333333333333333);
+  ERROR("Failed to trigger PARSE_ERROR in a parse error test");
+}
+
+// [[OutputRegex, In string:.*At line 2 column 16:.Value -0.8 is below the
```

Comment:
@wthrowe what are your thoughts on testing the bounds? Is this valuable or is it just testing the option parsing code?

Reply:
Tough one, actually.  It does test that you've hooked into the option parsing code correctly (not misspelled `lower_bound` or something), so the test isn't worthless.  On the other hand, as you've said many times test overhead is large, so is it really worth the time to check that the bounds are specified correctly when each such check requires a full test case (and there tend to be a lot of them)?

My feeling is that it is not worth it.  The documentation and sanity checks these bounds add are good, but it's not a major issue if they are missing.  It is unlikely that there will be a regression breaking the bounds (except in the parser, but that will be caught elsewhere).

Reply:
Okay, thanks! So then let's avoid these tests so as to keep our test runtime down :)